src/services/account/hooks/index.js

```diff
@@ -260,8 +280,16 @@ const restrictToUsersSchool = async (context) => {
 	return context;
 };
 
+const validateUserName = async (context) => {
```

Comment:
please check that this works with the usernames generated by LDAP and TSP, eg. 
services/sync/strategies/ldapschoolsyncer line 122
services/sync/strategies/TSP/TSP.js line 42

Reply:
Ok, I think I have fixed that.
I am checking if systemId exists. If it is then username is not checked because it may have many different formats if I understand it correctly. I wanted to add unit tests for this but I am getting some error because credentialCheck function is not available in LdapStrategy. It is available although in IservStrategy and MoodleStrategy
![image](https://user-images.githubusercontent.com/5371568/93481760-35222300-f8ff-11ea-842e-16346e8b9157.png)

I have added this to backup/setup/systems.json but I am getting error, which is presented above
```
  {
    "_id": {
      "$oid": "5f63485f8047a6303463b99b"
    },
    "type": "ldap",
    "alias": "LDAP School",
    "__v": 0
  }
```